src/ffi.rs

```diff
@@ -260,6 +260,49 @@ extern "C" {
 }
 
 
+#[cfg(feature = "std")]
+#[no_mangle]
+/// A reimplementation of the C function `secp256k1_context_create` in rust.
+///
+/// This function allocates memory, the pointer should be deallocated using `secp256k1_context_destroy`
+/// A failure to do so will result in a memory leak.
+///
+/// This will create a secp256k1 raw context.
+// Returns: a newly created context object.
+//  In:      flags: which parts of the context to initialize.
+pub unsafe extern "C" fn secp256k1_context_create(flags: c_uint) -> *mut Context {
+    assert!(mem::align_of::<usize>() >= mem::align_of::<u8>());
+
+    let size: usize =  secp256k1_context_preallocated_size(flags)/mem::size_of::<usize>();
+
+    let buf = vec![0usize; size + 1].into_boxed_slice();
+    let ptr = Box::into_raw(buf) as *mut usize;
+    ::core::ptr::copy_nonoverlapping(&size, ptr, 1);
+    let ptr: *mut usize = ptr.offset(1);
+
+    secp256k1_context_preallocated_create(ptr as *mut c_void, flags)
+}
+
+#[cfg(feature = "std")]
+#[no_mangle]
+/// A reimplementation of the C function `secp256k1_context_destroy` in rust.
+///
+/// This function destroys and deallcates the context created by `secp256k1_context_create`.
+///
+/// The pointer shouldn't be used after passing to this function, consider it as passing it to `free()`.
+///
+pub unsafe extern "C" fn secp256k1_context_destroy(ctx: *mut Context) {
+    assert_eq!(ctx as usize % mem::align_of::<usize>(), 0);
+    secp256k1_context_preallocated_destroy(ctx);
+    let ctx: *mut usize = ctx as *mut usize;
+
+    let size_ptr: *mut usize = ctx.offset(-1);
+    let size: usize = ::core::ptr::read(size_ptr);
+    let slice: &mut [usize] = slice::from_raw_parts_mut(size_ptr , size+1);
```

Comment:
Please rename `size` to `n_words` and drop the `+ 1`

Reply:
The +1 is important because that size doesn't account for itself.
and if we want the memory to be dropped we need the box to be reconstructed.
although I'm wondering now if the `ptr::read()` that memory be dropped afterwards